django/views/debug.py

```diff
@@ -544,6 +545,54 @@ def get_exception_traceback_frames(self, exc_value, tb):
                 pre_context = []
                 context_line = "<source code not available>"
                 post_context = []
+            colno = ""
+            if PY311:
+                _, _, start_column, end_column = next(
+                    itertools.islice(
+                        tb.tb_frame.f_code.co_positions(), tb.tb_lasti // 2, None
+                    )
+                )
+                if start_column and end_column:
+                    from traceback import (
+                        StackSummary,
+                        _byte_offset_to_character_offset,
+                        _extract_caret_anchors_from_line_segment,
+                        _walk_tb_with_full_positions,
+                    )
```

Comment:
move imports to module level

Reply:
IMO it's too much, it's not worth the additional complexity and adding so many Python's internals to Django.

Reply:
Yea I agree with @felixxm.
This is a case that shouldn't even occur on a production server, so it is better to add as few imports as possible.

Reply:
@adamchainz Does it work for you? I think we should revert recent changes and only use `start_column`, `end_column` to highlight the general error location. At least until an official API is added to Python.

Reply:
I agree. Perhaps we could open an upstream github issue to ask about this?

Reply:
@felixxm @adamchainz Alright, no worries. Now only the general error location gets highlighted with `start_column` and `end_column`. I've also fixed unicode behaviour. Please check again :) Let's wait until the official API gets updated.